warehouse/internal/repo/staging.go

```diff
@@ -401,22 +401,41 @@ func (sf *StagingFiles) TotalEventsForUpload(ctx context.Context, upload model.U
 	return total.Int64, nil
 }
 
-func (sf *StagingFiles) FirstEventForUpload(ctx context.Context, upload model.Upload) (time.Time, error) {
-	var firstEvent sql.NullTime
-	err := sf.db.QueryRowContext(ctx, `
+func (sf *StagingFiles) GetEventTimeRangesByUploadID(ctx context.Context, uploadID int64) ([]*model.EventTimeRange, error) {
```

Comment:
Do you really need to return a slice of pointers? Wouldn't it make more sense to return a slice of values instead? It should be easier on the GC.